.github/workflows/e2e-infra-deploy.yml

```diff
@@ -0,0 +1,70 @@
+# PROCESS
+#
+# 1. Deploy the core stack using AWS CDK.
+# 2. Deploy the AOT stack using AWS CDK.
+
+# USAGE
+#
+# This workflow is used by the E2E Tests workflow to deploy the necessary infrastructure.
+
+name: E2E Tests Infra deployment
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  deploy-core-stack:
+    runs-on: aws-powertools_ubuntu-latest_8-core
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.E2E_DEPLOY_ROLE }}
+          aws-region: us-east-1
+
+      - name: Set up .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: '8.x'
+
+      - name: Install CDK
+        run: npm install -g aws-cdk
+
+      - name: Deploy Core Stack
+        run: |
+          cd libraries/tests/e2e/infra
+          cdk deploy --require-approval never
+  
+  deploy-aot-stack:
+    runs-on: aws-powertools_ubuntu-latest_8-core
+    strategy:
+      matrix:
+        architecture: [ x86_64, arm64 ]
+    container:
+      image: public.ecr.aws/sam/build-dotnet8:latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
```

Comment:
As a best practices, please put the sha256 that matches this version. Dependabot will bump it.